src/nvim/main.c

```diff
@@ -293,7 +276,8 @@ int main(int argc, char **argv)
   if (params.diff_mode)
     diff_win_options(firstwin, FALSE);
 
-  cmdline_row = Rows - p_ch;
+  assert(p_ch >= 0 && Rows >= p_ch && Rows - p_ch <= INT_MAX);
+  cmdline_row = (int)Rows - p_ch;
```

Comment:
The assert is wrong. I wrote `(int)(Rows - p_ch)`, not `(int)Rows - p_ch`.
What you have written behaves this way:
- Rows is converted to int.
- Before doing substraction, as you have an int and a long (p_ch), the int gets promoted to long again.
- The substraction is done.
- Result of the substraction (long) is going to be assigned to an int var (cmdline_row), loosing precision. Thus the warning.